quartz/components/PageTitle.tsx

```diff
@@ -1,21 +1,71 @@
 import { pathToRoot } from "../util/path"
 import { QuartzComponentConstructor, QuartzComponentProps } from "./types"
 import { classNames } from "../util/lang"
+import * as fs from "fs"
+import * as path from "path"
 
-function PageTitle({ fileData, cfg, displayClass }: QuartzComponentProps) {
-  const title = cfg?.pageTitle ?? "Untitled Quartz"
-  const baseDir = pathToRoot(fileData.slug!)
-  return (
-    <h1 class={classNames(displayClass, "page-title")}>
-      <a href={baseDir}>{title}</a>
-    </h1>
-  )
+interface Options {
+  iconFolderPath: string
+  defaultIcon?: string
 }
 
-PageTitle.css = `
-.page-title {
-  margin: 0;
+const defaultOpts: Options = {
+  iconFolderPath: "",
 }
-`
 
-export default (() => PageTitle) satisfies QuartzComponentConstructor
+export default ((userOpts?: Partial<Options>) => {
+  const opts = { ...defaultOpts, ...userOpts }
+  function PageTitle({ fileData, cfg, displayClass }: QuartzComponentProps) {
+    const title = cfg?.pageTitle ?? "Untitled Quartz"
+    const iconType = fileData.frontmatter?.icon ?? opts.defaultIcon
+    const baseDir = pathToRoot(fileData.slug!)
+    if (opts.iconFolderPath.trim().length === 0 || !iconType) {
+      return (
+        <h1 class={classNames(displayClass, "page-title")}>
+          <a href={baseDir}>{title}</a>
+        </h1>
+      )
+    }
+    const iconFullPath = `${opts.iconFolderPath}/${iconType || opts.defaultIcon}.svg`
+    let iconAsSVG = ""
+    try {
+      iconAsSVG = fs.readFileSync(path.join(process.cwd(), iconFullPath), "utf8")
+    } catch (e) {
+      iconAsSVG = fs.readFileSync(
+        path.join(process.cwd(), `${opts.iconFolderPath}/${opts.defaultIcon}.svg`),
+        "utf8",
+      )
+    }
+    return (
+      <div
+        class={classNames(displayClass, "page-title")}
+        data-icon={iconFullPath}
+        data-hasIcon={true}
+      >
+        {iconAsSVG && (
+          <div class="page-title-icon" dangerouslySetInnerHTML={{ __html: iconAsSVG }} />
```

Comment:
ughh...

probably safe right ? 😄

Reply:
Well, I don't found any other way to set this icon in safe manner :(

Reply:
Also, I tried to replace the folder button to collapse by one of the icon but it's seems impossible, as I said in discord earlier :<

Reply:
for my callout icon PR, i struggled with the same issue. So i say that svg icons are sended by css. 
Maybe we could have a similar approach here

Reply:
The goal was to make it compatible with the plugin iconize so... It will just make the thing useless :/

Reply:
The only thing I can propose here is to use css to tweaks like a cursed cake